src/components/dom.ts

```diff
@@ -349,27 +349,20 @@ export default class Dom {
    *
    * @returns {boolean}
    */
-  public static canSetCaret(target: HTMLElement): boolean {
-    let result = true;
+  public static canSetCaret(target: HTMLElement | Node): boolean {
+    const selectionSupportedTypes = [
+      'text',
+      'search',
+      'tel',
+      'url',
+      'password',
+    ];
```

Comment:
Looks like some types are missed:

- email
- number
- month
- week

And maybe these, not sure:

- date
- datetime-local
- time

Reply:
Only the five defined types support selection. Here's [HTML specs](https://html.spec.whatwg.org/multipage/input.html#concept-input-apply:~:text=%C2%B7-,selectionStart,-%C2%B7) and [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setSelectionRange#:~:text=text%2C%20search%2C%20URL%2C%20tel%20and%20password) for it

Reply:
Good idea, done now!